### PR TITLE
Make Cesium work with current release of gdal2tiles.py

### PR DESCRIPTION
**Pull request by mramato**
gdal2tiles.py previously had 2 bugs compared to the TMS specification.

1. Origin and BoundingBox flipped X and Y values.
2. Origin and Bounding box were always in geodetic degrees, regardless of the file.

They fixed 1, but 2 is still an issue. Our workarounds were kind of smashed together in the code so I detangled it to remove doing 1 by default, but added a `flipXY` property for manually enabled backwards compatibility. Unfortunately, the buggy and non buggy versions of gdal generated identical files except for the flipped values, so it's impossible for us to detect this ourselves.

I also fixed bad promise resolution in tilemapresource.xml, which would silently fail (in the original code) when `flipXY` was performed but no longer needed. Fixing this makes some buggy TMS sets load properly anyway, so that's a win for us.

For TMS layers we have in the repository, I fixed them to not contain the bug.  Only tests that specifically check for `flipXY` behavior include the buggy XML.

I did NOT change the black marble data and instead added `flipXY` to the Sandcastle example and defaultLayers code.  We can certainly regenerate it if we want, but it will break old versions of Cesium that are currently using it.

Turns out, this issue was actually reported on the forum back in October, but we all missed it.  I didn't notice that post until I ran into the bug myself and went digging into the problem.

I'd really like to get this into 1.18 on Monday and talked to @shunter about these changes already.  If @shunter or @kring could review and merge this, that would be great. Thanks!

### CODE REVIEW

File: Source/Scene/createTileMapServiceImageryProvider.js [hunk 1]
```diff
@@ -175,30 +167,30 @@ define([
...
+                // Determine based on the profile attribute if this tileset was generated by gdal2tiles.py, which
+                // uses 'mercator' and 'geodetic' profiles, or by a tool compliant with the TMS standard, which is
+                // 'global-mercator' and 'global-geodetic' profiles.
+                //
+                // In the gdal2Tiles case, X and Y are always in geodetic degrees. In older versions of gdal (and only,
+                // gdal) they were also flipped, which is why we check for an option to flip the values here as well.
+                // Unfortunately there is no way to autodetect whether flipping is needed or not.
+                var isGdal2tiles = tilingSchemeName === 'geodetic' || tilingSchemeName === 'mercator';
+                var flipXY = defaultValue(options.flipXY, false);
+                if (isGdal2tiles && flipXY) {
```
Comment (kring) [inline Source/Scene/createTileMapServiceImageryProvider.js]: If you're going to provide an `options.flipXY`, you should make it flip all the time, not just for a tileset that was apparently produced by gdal2tiles.py.  Or, at least, the doc needs to be very clear that the option is ignored if we don't think it came from gdal2tiles.py.

**Comment by kring** [comment]
This looks good to me, other than the one comment.  Sorry I didn't get a chance to review it before 1.18.

**Comment by mramato** [comment]
No worries.  Thanks for the review.

It turns out there are still some issues.  For one, using the `geodetic` profile seems to generate tilesets that are unusable in Cesium.  Should this be the case?  I was under the assumption (perhaps incorrectly) that geodetic would be the preferred profile because it avoid reprojection and the tiling scheme seems to be properly detected.  It just requests wildly different tiles (at one point it looked like to be that the z level ended up being off by 1, but I haven't dug to deeply into it yet.  Any thoughts here?

Now that 1.18 is out, I'm in no rush to merge this.  They just released a new gdal as well, so I want to grab that and make sure Cesium can handle everything properly.

File: Source/Scene/createTileMapServiceImageryProvider.js (hunk 1 above)
Comment (mramato) [inline Source/Scene/createTileMapServiceImageryProvider.js]: Fair enough, there's no reason for the user to specify it unless they know it needs to be flipped, so we might as well change the code to flip all of the time if asked.

**Comment by kring** [comment]
> For one, using the geodetic profile seems to generate tilesets that are unusable in Cesium. Should this be the case?

Yeah geodetic _should_ work in Cesium, and it will work at the poles and it will let Cesium skip reprojection, like you said.  I wouldn't be shocked if geodetic tile generation were just completely broken in gdal2tiles.py, since it's mostly used with web maps which are all Web Mercator based.  I also wouldn't be shocked to see a bug in Cesium causing the problem.  Keep in mind that in the `GeographicTilingScheme`, the root of the tile pyramid has two tiles, the Western hemisphere and the Eastern hemisphere.

**Comment by mramato** [comment]
Okay, my changes are in.  I'll write up a separate issue to look into geodetic tilesets when we have the time.

**Comment by mramato** [comment]
@kring can you please merge this unless you have any other feedback.  Thanks.

**Pull request merged**